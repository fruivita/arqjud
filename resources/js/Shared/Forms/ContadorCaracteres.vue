<!--
    Contador de caracteres para os elementos de formulÃ¡rio.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->

<script setup>
import { computed } from 'vue';

const props = defineProps({
    texto: { type: String, required: true },
    maxlength: { type: [Number, String], required: true },
});

const contador = computed(() => props.texto?.length ?? 0);
</script>

<template>
    <div class="absolute inset-y-0 right-0 flex items-center pr-3 text-sm">
        <span dusk="contador">{{ contador }}</span
        >/<span dusk="maxlength">{{ maxlength }}</span>
    </div>
</template>
