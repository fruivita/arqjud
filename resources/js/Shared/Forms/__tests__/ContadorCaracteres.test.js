/**
 * Testes para o componente ContadorCaracteres.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import ContadorCaracteres from '@/Shared/Forms/ContadorCaracteres.vue';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;

beforeEach(() => {
    mountFunction = (options = {}) => {
        return mount(ContadorCaracteres, { ...options });
    };
});

// Caminho feliz
describe('ContadorCaracteres', () => {
    test('propriedades do componente estÃ£o definidas', () => {
        expect(ContadorCaracteres.props).toMatchObject({
            texto: { type: String, required: true },
            maxlength: { type: [Number, String], required: true },
        });
    });

    test('renderiza o componente respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { texto: 'foo', maxlength: 5 },
            }).html()
        ).toMatchSnapshot();
    });

    const cases = [
        ['', '0'],
        ['foo', '3'],
        ['foobar', '6'],
    ];

    test.each(cases)('exibe a quantidade de caracteres digitada', (entrada, saida) => {
        const wrapper = mountFunction({
            props: { texto: entrada, maxlength: 5 },
        });

        expect(wrapper.get('[dusk="maxlength"]').text()).toBe('5');
        expect(wrapper.get('[dusk="contador"]').text()).toBe(saida);
    });
});
