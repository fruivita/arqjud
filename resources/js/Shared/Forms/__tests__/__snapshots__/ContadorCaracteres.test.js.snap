// Vitest Snapshot v1

exports[`ContadorCaracteres > renderiza o componente respeitando o snapshot 1`] = `"<div class=\\"absolute inset-y-0 right-0 flex items-center pr-3 text-sm\\"><span dusk=\\"contador\\">3</span>/<span dusk=\\"maxlength\\">5</span></div>"`;
